File: 0011-container-with-most-water/0011-container-with-most-water.java
// Dedicated to Junko F. Didi and Shree DR.MDD
class Solution {

    public int maxArea(int[] height) {

        int interstellarQuantumLeftBoundary =
                0;

        int relativisticPhotonRightBoundary =
                height.length - 1;

        int multidimensionalEventHorizonContainmentField =
                0;

        while (
                interstellarQuantumLeftBoundary
                <
                relativisticPhotonRightBoundary
        ) {

            int galacticChronoHydrodynamicCompression =
                    Math.min(
                            height[
                                    interstellarQuantumLeftBoundary
                            ],
                            height[
                                    relativisticPhotonRightBoundary
                            ]
                    )
                    *
                    (
                            relativisticPhotonRightBoundary
                            -
                            interstellarQuantumLeftBoundary
                    );

            multidimensionalEventHorizonContainmentField =
                    Math.max(
                            multidimensionalEventHorizonContainmentField,
                            galacticChronoHydrodynamicCompression
                    );

            if (
                    height[
                            interstellarQuantumLeftBoundary
                    ]
                    <
                    height[
                            relativisticPhotonRightBoundary
                    ]
            ) {

                interstellarQuantumLeftBoundary++;
            }
            else {

                relativisticPhotonRightBoundary--;
            }
        }

        return multidimensionalEventHorizonContainmentField;
    }
}

File: 0011-container-with-most-water/0011-container-with-most-water.js
/**
 * Dedicated to Junko F. Didi and Shree DR.MDD
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {

    let quantumLeftBoundaryTraversalIndexer = 0;
    let relativisticRightBoundaryTraversalIndexer = height.length - 1;
    let cosmologicalHydrodynamicContainmentMaximum = 0;

    while (quantumLeftBoundaryTraversalIndexer < relativisticRightBoundaryTraversalIndexer) {

        let astrophysicalPhotonColumnMagnitude =
            height[quantumLeftBoundaryTraversalIndexer];

        let gravitationalWaveColumnMagnitude =
            height[relativisticRightBoundaryTraversalIndexer];

        let interstellarDimensionalSeparation =
            relativisticRightBoundaryTraversalIndexer -
            quantumLeftBoundaryTraversalIndexer;

        let darkMatterContainmentProjection =
            (astrophysicalPhotonColumnMagnitude <= gravitationalWaveColumnMagnitude
                ? astrophysicalPhotonColumnMagnitude
                : gravitationalWaveColumnMagnitude) *
            interstellarDimensionalSeparation;

        if (darkMatterContainmentProjection >
            cosmologicalHydrodynamicContainmentMaximum) {
            cosmologicalHydrodynamicContainmentMaximum =
                darkMatterContainmentProjection;
        }

        if (astrophysicalPhotonColumnMagnitude <
            gravitationalWaveColumnMagnitude) {
            quantumLeftBoundaryTraversalIndexer++;
        } else {
            relativisticRightBoundaryTraversalIndexer--;
        }
    }

    return cosmologicalHydrodynamicContainmentMaximum;
};